src/internal.c

```diff
@@ -729,15 +703,28 @@ static int ImportKeyState(WOLFSSL* ssl, byte* exp, word32 len, byte ver)
     keys->encryptionOn = exp[idx++];
     keys->decryptedCur = exp[idx++];
 
-#ifdef WORD64_AVAILABLE
-    ato64(exp + idx, &keys->window);     idx += OPAQUE64_LEN;
-    ato64(exp + idx, &keys->prevWindow); idx += OPAQUE64_LEN;
-#else
-    ato32(exp + idx, &keys->window);     idx += OPAQUE32_LEN;
-    ato32(exp + idx, 0);                 idx += OPAQUE32_LEN;
-    ato32(exp + idx, &keys->prevWindow); idx += OPAQUE32_LEN;
-    ato32(exp + idx, 0);                 idx += OPAQUE32_LEN;
-#endif
+    {
+        word16 windowSz, i, adj = 0;
+
+        ato16(exp + idx, &windowSz);
+        idx += OPAQUE16_LEN;
+
+        if (windowSz > WOLFSSL_DTLS_WINDOW_WORDS) {
```

Comment:
I wasn't taking into account the word size for the adj value.

But, if the stored window size is greater than what the importer expects, the importer caps the size at its maximum, and only copies so many words. adj skips the remainder in the store. If the importer expects more, it'll get to copy what it can and move on.

But, this means the importer expects more than it gets, it should mark the unset remainder to "already received".